### PR TITLE
AB#87724 Build activity type

### PR DESCRIPTION
**Pull request by Carq**
image

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs [hunk 1]
```diff
@@ -73,8 +71,6 @@ public class DeliveryPhaseEntity : IDeliveryPhaseEntity
 
     public CompletionMilestoneDetails? CompletionMilestone { get; private set; }
 
-    public IsAdditionalPaymentRequested? IsAdditionalPaymentRequested { get; private set; }
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs]: why this was removeed?

File: src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs [hunk 2]
```diff
@@ -105,26 +101,21 @@ public void SetHomesToBeDeliveredInThisPhase(IReadOnlyCollection<HomesToDeliverI
         }
     }
 
-    public void ProvideName(DeliveryPhaseName deliveryPhaseName)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs]: why this was removeed?

File: src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs [hunk 3]
```diff
@@ -105,26 +101,21 @@ public void SetHomesToBeDeliveredInThisPhase(IReadOnlyCollection<HomesToDeliverI
...
-    {
-        Name = _modificationTracker.Change(Name, deliveryPhaseName, MarkAsNotCompleted);
-    }
-
     public void ProvideAcquisitionMilestoneDetails(AcquisitionMilestoneDetails? details)
     {
-        if (Organisation.IsUnregisteredBody)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Delivery/Entities/DeliveryPhaseEntity.cs]: why this was renamed?

File: src/HE.Investment.AHP.WWW/Workflows/DeliveryPhaseWorkflow.cs [hunk 4]
```diff
@@ -35,6 +35,7 @@ public bool CanBeAccessed(DeliveryPhaseWorkflowState state)
             DeliveryPhaseWorkflowState.Create => true,
             DeliveryPhaseWorkflowState.Name => true,
             DeliveryPhaseWorkflowState.TypeOfHomes => true,
+            DeliveryPhaseWorkflowState.BuildActivityType => true,
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Workflows/DeliveryPhaseWorkflow.cs:38]: please update tests

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    BuildActivityTypeOptions.BuildActivityTypeForNewBuildOptions(Model.BuildActivityTypeForNewBuild);
+}
+
+<gds-div-grid-column-two-thirds>
+    <vc:workflow-back-button action="@DeliveryPhaseWorkflowState.BuildActivityType.ToString()" param-name="deliveryPhaseId"/>
+
+    <form asp-controller="DeliveryPhase" asp-action="BuildActivityType" asp-method="post" novalidate>
+        <vc:form-header caption="@($"{Model.ApplicationName} - {Model.Name}")" title="@title"/>
+        @Html.HiddenFor(m => m.TypeOfHomes)
+        <gds-field-set role="group">
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml:42]: why fieldsset is needed here?

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Controllers/DeliveryPhaseController.cs [hunk 6]
```diff
@@ -84,14 +83,7 @@ public async Task<IActionResult> Name([FromRoute] string applicationId, [FromRou
         return await Continue(new { applicationId, deliveryPhaseId });
     }
 
-    [HttpPost("{deliveryPhaseId}/Name")]
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Controllers/DeliveryPhaseController.cs:87]: Are you sure about removing this?

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml (hunk 5 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW/Views/DeliveryPhase/BuildActivityType.cshtml:42]: Removed.

File: src/HE.Investment.AHP.WWW/Workflows/DeliveryPhaseWorkflow.cs (hunk 4 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW/Workflows/DeliveryPhaseWorkflow.cs:38]: Done.

**Pull request merged**